src/ansys/dynamicreporting/core/utils/report_objects.py

```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
-    def predictor_variables(self, value):
-        props = self.get_property()
-        props["predictor_variables"] = value
-        self.set_property(props)
+    def set_table_name(self, value=""):
+        if type(value) is not str:
+            raise ValueError("Error: input should be a string")
+        d = json.loads(self.params)
+        if "stats_params" not in d:
+            d["stats_params"] = {}
+        d["stats_params"]["table_name"] = value
+        self.params = json.dumps(d)
+        return
 
-    @property
-    def response_variables(self):
-        return self.get_property().get("response_variables")
+    def get_tree_name(self):
+        if "stats_params" in json.loads(self.params):
+            if "tree_name" in json.loads(self.params)["stats_params"]:
+                return json.loads(self.params)["stats_params"]["tree_name"]
+        return ""
 
-    @response_variables.setter
-    def response_variables(self, value):
-        props = self.get_property()
-        props["response_variables"] = value
-        self.set_property(props)
+    def set_tree_name(self, value=""):
+        if type(value) is not str:
+            raise ValueError("Error: input should be a string")
+        d = json.loads(self.params)
+        if "stats_params" not in d:
+            d["stats_params"] = {}
+        d["stats_params"]["tree_name"] = value
+        self.params = json.dumps(d)
+        return
 
-    @property
-    def intercept(self):
-        return self.get_property().get("intercept")
+    def get_predictor_variables(self):
+        if "stats_params" in json.loads(self.params):
+            if "predictor_variables" in json.loads(self.params)["stats_params"]:
+                return json.loads(self.params)["stats_params"]["predictor_variables"]
+        return []
 
-    @intercept.setter
-    def intercept(self, value):
-        props = self.get_property()
-        props["intercept"] = value
-        self.set_property(props)
+    def set_predictor_variables(self, value):
+        if type(value) is not list:
+            raise ValueError("Error: input should be an array")
+        # standard input format is a 2d array with a subarray length of 3
+        if len(value[0]) != 3:
+            raise ValueError(
+                "Error: input format should be an array of subarrays each of length 3. With Type, Predictor, Output Name."
+            )
+        d = json.loads(self.params)
+        if "stats_params" not in d:
+            d["stats_params"] = {}
+        d["stats_params"]["predictor_variables"] = json.dumps(value)
+        self.params = json.dumps(d)
+        return
+
+    def get_response_variables(self):
+        if "stats_params" in json.loads(self.params):
+            if "response_variables" in json.loads(self.params)["stats_params"]:
+                return json.loads(self.params)["stats_params"]["response_variables"]
+        return ""
+
+    def set_response_variables(self, value=""):
+        if type(value) is not list:
```

Comment:
`if not isinstance(value, list)`

Reply:
Hello Visesh, thank you so much your suggestion, will update this part